tools/mksurfdata_map/src/mkpftMod.F90

```diff
@@ -509,24 +513,21 @@ subroutine mkpft(ldomain, mapfname, fpft, ndiag, allow_no_crops, &
 
      call domain_checksame( tdomain, ldomain, tgridmap )
 
-     ! Area-average percent cover on input grid [pctpft_i] to output grid 
-     ! [pctpft_o] and correct [pctpft_o] according to land landmask
-     ! Note that percent cover is in terms of total grid area.
-  
-     do no = 1,ns_o
```

Comment:
I'm thinking you could keep this code here if you followed my above suggestion about making a separate routine that calculates wtnorm (i.e., frac_dst). Then you could call that routine first, then do this code. That would be preferable to recalculating pctlnd_o for every pft, as you now do.